libs/vkd3d/resource.c

```diff
@@ -1211,6 +1212,132 @@ HRESULT vkd3d_get_image_allocation_info(struct d3d12_device *device,
     return hr;
 }
 
+struct vkd3d_view_key
+{
+    enum vkd3d_view_type view_type;
+    union
+    {
+        struct vkd3d_buffer_view_desc buffer;
+        struct vkd3d_texture_view_desc texture;
+    } u;
+};
+
+struct vkd3d_view_entry
+{
+    struct hash_map_entry entry;
+    struct vkd3d_view_key key;
+    struct vkd3d_view *view;
+};
+
+uint32_t vkd3d_view_entry_hash(const void *key)
+{
+    const struct vkd3d_view_key *k = key;
+    uint64_t resource_handle;
+    uint32_t hash;
+
+    switch (k->view_type)
+    {
+        case VKD3D_VIEW_TYPE_BUFFER:
+            resource_handle = (uint64_t)k->u.buffer.buffer;
+            hash = hash_combine(resource_handle, resource_handle >> 32);
```

Comment:
Should probably cast the resource_handle to uint32_t for consistency with the other casting below?

Reply:
Ended up fixing this differently by adding a simple hash function for 64-bit numbers.